contribute/design/proposal-cstor-orchestration.md

```diff
@@ -0,0 +1,311 @@
+         type: block
+         node: node-host-label
+       spec:
+         scope: Cluster
+         disks:
+           #PDCs as mentioned in the SPC. 
+           pdc-list: ["pdc-name1", "pdc-name2",...]
+           #PDs that are associated with this Pool.
+           pd-list: ["pd-name1", "pd-name2",...]
+           #This can contain the list of PDs that can be used as hot-spares if any of the
+           # used disk are errored.
+           #pd-list-spares: ["pd-namex", "pd-namey",...]
+         poolspec: 
+           #Defines the type of pool as passed from the SPC. stripe or mirror. 
+           type: "stripe"
+           #Save the resources as sent from SPC. If these values are updated, 
+           # the corresponding values on the cStorPool Deployment YAML will have to be updated.
+           #resources:
+             #cpu: 
+             #memory: 
+           #Pool features as passed from the SPC.
+           #thinprovisioning: true       
+       ```
+   * Admin associates the SPC with a StorageClass
+     ```
+     apiVersion: storage.k8s.io/v1
+     kind: StorageClass
+     metadata:
+       name: openebs-pool1
+     provisioner: openebs.io/provisioner-iscsi
+     parameters:
+       openebs.io/storage-pool-claim: pool1
+     ```
+
+2. Creating Volume using the (cStor) Storage Pools:
+
+   * Admin will create a PVC that is associated to StoragePool (linked by the pool name). 
+     ```
+     apiVersion: v1
+     kind: PersistentVolumeClaim
+     metadata:
+       name: demo-vol
+     spec:
+       accessModes:
+         - ReadWriteOnce
+       resources:
+         requests:
+           storage: 5G
+       storageClassName: openebs-pool1
+     ```
+
+   * maya-apiserver will create the OV cStorService (to get the portal ip address) and the OV cStorController. The OV cStorController will be passed a ConfigMap - cStorControllerConfig (that will be mounted into the OV cStorController). 
+     - The cStorService YAML will have the following details:
+       ```
+       apiVersion: v1
+       kind: Service
+       metadata:
+         name: pvc-ee171da3-07d5-11e8-a5be-42010a8001be-cstor-ctrl-service
+         labels:
+           pv: pvc-ee171da3-07d5-11e8-a5be-42010a8001be
+       spec:
+         type: ClusterIP
+         ports:
+         - name: iscsi
+           port: 3260
+           protocol: TCP
+           targetPort: 3260
+         selector:
+           openebs/controller: cstor-controller
+           pv: pvc-ee171da3-07d5-11e8-a5be-42010a8001be
+       ```
+       
+     - The cStorControllerConfigMap will contain the following details:
+       - portal ip - obtained from the cStorService ClusterIP
+       - volume id - will be pv id (like ee171da3-07d5-11e8-a5be-42010a8001be), used from the pvc name. 
+       - volume name - will also be the pv id. _(TODO Or should this be pvc name like demo-vol. Verify this with @gila)_
+       - config-map name - pvc-ee171da3-07d5-11e8-a5be-42010a8001be-cstor-ctrl-config-map
```

Comment:
should we consider replacing config-map with CRD here also @kmova ?

Reply:
changed it..